samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -278,6 +282,7 @@ private static void calculateIntStreamPartitions(JobGraph jobGraph, Config confi
     }
     for (StreamEdge edge : jobGraph.getIntermediateStreamEdges()) {
       if (edge.getPartitionCount() <= 0) {
+        log.info("Set the partition count for join input {} to {}.", edge.getName(), partitions);
```

Comment:
Regarding log message:

1. `edge` is not necessarily — and probably not likely to be — join input.
2. My preference if I'm looking at logs produced by `ExecutionPlanner` is to have this message be different from the message logged on line 220 so that it's easy to discern *why* a partition count value was set any given value w/o having to remember to look at the method name where each message was logged.

I suggest we change:
1. this log message to something like: `"Set partition count of intermediate stream edge <edge-name> to <value>."`
2. the log message on line 220 to something like: `"Set partition count of intermediate stream edge <edge-name> participating in join <join-op-id> to <value>."`